administrator/components/com_jedchecker/libraries/rules/language.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * @package    Joomla.JEDChecker
+ *
+ * @copyright  Copyright (C) 2021 Open Source Matters, Inc. All rights reserved.
+ *
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die('Restricted access');
+
+
+// Include the rule base class
+require_once JPATH_COMPONENT_ADMINISTRATOR . '/models/rule.php';
+
+
+/**
+ * class JedcheckerRulesLanguage
+ *
+ * This class validates language ini file
+ *
+ * @since  2.3
+ */
+class JedcheckerRulesLanguage extends JEDcheckerRule
+{
+	/**
+	 * The formal ID of this rule. For example: SE1.
+	 *
+	 * @var    string
+	 */
+	protected $id = 'LANG';
+
+	/**
+	 * The title or caption of this rule.
+	 *
+	 * @var    string
+	 */
+	protected $title = 'COM_JEDCHECKER_LANG';
+
+	/**
+	 * The description of this rule.
+	 *
+	 * @var    string
+	 */
+	protected $description = 'COM_JEDCHECKER_LANG_DESC';
+
+	/**
+	 * Initiates the search and check
+	 *
+	 * @return    void
+	 */
+	public function check()
+	{
+		// Find all INI files of the extension (in the format tag.extension.ini or tag.extension.sys.ini)
+		$files = JFolder::files($this->basedir, '^[a-z]{2,3}-[A-Z]{2}\.\w+(?:\.sys)?\.ini$', true, true);
+
+		// Iterate through all the ini files
+		foreach ($files as $file)
+		{
+			// Try to validate the file
+			$this->find($file);
+		}
+	}
+
+	/**
+	 * Reads and validates an ini file
+	 *
+	 * @param   string  $file  - The path to the file
+	 *
+	 * @return bool True on success, otherwise False.
+	 */
+	protected function find($file)
+	{
+		$lines = file($file);
+		foreach ($lines as $lineno => $line)
+		{
+			$line = trim($line);
+
+			// Check for BOM sequence
+			if ($lineno === 0 && strncmp($line, "\xEF\xBB\xBF", 3) === 0)
+			{
+				$this->report->addError($file, JText::_('COM_JEDCHECKER_LANG_BOM_FOUND'), 1);
+				// Romeve BOM for further checks
+				$line = substr($line, 3);
+			}
+
+			// Skip empty lines, comments, and section names
+			if ($line === '' || $line[0] === ';' || $line[0] === '[')
+			{
+				continue;
+			}
+
+			// Report incorrect comment character
+			if ($line[0] === '#')
+			{
+				$this->report->addError($file, JText::_('COM_JEDCHECKER_LANG_INCORRECT_COMMENT') .
+					'<br>' . htmlspecialchars($line), $lineno);
+				continue;
+			}
+
+			// Check for "=" character in the line
+			if (strpos($line, '=') === false)
```

Comment:
There has been many languages files that by mistake has line brakes in the language strings... and this will cause this test to be a false positive. Yes it is not good practice... but those files actually still work. So a **warning** with direction to remove all line breaks in the language strings is better.

Reply:
There is a commit for this case too: https://github.com/dryabov/jedchecker/commit/eb60ee745594e908fa0ea5017e9ff47cb0c40ea1
But note that it renames variable $lineno to $startLineno (to refer to the first line of multi-line language string in the error/warning message), and as a result it's better to merge it after other language.php-related commits are merged (otherwise all related merges should be made manually).

Reply:
I will make a note of this... as I will keep this PR in develop branch until the above mentioned PR is also there, to correct the issue. Yet I know from experience the removing of all line breaks `\n` from the actually language value is better as it resolves possible system errors. So we should give a warning here... and encourage this to be changed.

Reply:
But we should not block the component from the JED, that is why I said not to give an error here. As errors are serious, and prevents components from being added to the index... lets remember that. We rather give warnings to direct good convention... and only errors where it is 100% not allowed, or 100% wrong.

Reply:
I've cherry-picked all commits from my jed-v3 branch, so all PRs are in "final" state. But firstly merge PRs #85 and #103, and then others may be merged in arbitrary order.

Reply:
The problem with legacy #-comments (supported in J!1.0/1.5 only) is that PHP doesn't support them as a comments and parses them as a key-value expression. As a result the following line will break all the file:
```
#TEST="
```
because PHP will try to found the end of that quoted multi-line string.
Note there is a requirement at https://docs.joomla.org/Specification_of_language_files: "you MUST use semicolons, otherwise your language files will not load".